docs/pages/upgrading/agent-managed-updates/bootstrap-ec2-agent.mdx

```diff
@@ -0,0 +1,249 @@
+the existing instances already part of the Teleport cluster. At the end of the joining process,
+the Auth Service signs certificates for the joining instance. Those certificates represent the
+trust that was established. With them, the newly-joined instance can interact with the other Teleport instances.
+
+For bootstrapping EC2, the most appropriate methods are creating short-lived or long-lived tokens,
+or using delegated join with the IAM join method.
+
+### Creating join tokens
+
+Create a join token with an explicit TTL and the right service type.
+By default, `tctl tokens add` issues short-lived tokens (expiry is 30 minutes if you omit --ttl).
+
+```code
+$ tctl tokens add --type=node --ttl=1h --format=text
+```
+
+Long-lived option: a token resource with an explicit expiry. If you need something that lasts longer than
+a few hours, prefer a token resource with a concrete `metadata.expires` timestamp instead of cranking
+`--ttl` to very large values. Example (set an appropriate, bounded expiry):
+
+```yaml
+# token-long-lived.yaml
+kind: token
+version: v2
+metadata:
+  name: node-join-24h
+  expires: 2025-12-31T23:59:00Z
+spec:
+  join_method: token
+  roles: [Node]
+```
+
+```code
+$ tctl create -f token-long-lived.yaml
+```
+
+### Delegated joins
+
+Delegated joins avoid shipping any secret token in user-data. You create a named token resource
+that encodes context rules (your AWS account, role ARNs, regions, etc.), then the agent proves
+its identity to the Auth Service using cloud-issued credentials.
+
+Your agent can fetch IAM credentials (e.g., EC2 instance profile, IRSA, or env vars).
+
+Create a joining token that allows your account (optionally constrain to a role ARN):
+
+```yaml
+# token-iam.yaml
+kind: token
+version: v2
+metadata:
+  name: iam-token
+spec:
+  roles: [Node]
+  join_method: iam
+  allow:
+    - aws_account: "111111111111"
+    # Optional: restrict to assumed-role pattern
+    - aws_account: "222222222222"
+      aws_arn: "arn:aws:sts::222222222222:assumed-role/teleport-node-role/i-*"
+```
+
+```code
+$ tctl create -f token-iam.yaml
+```
+
+```yaml
+teleport:
+  join_params:
+    method: iam
+    token_name: iam-token
+  proxy_server: example.teleport.sh:443
+ssh_service:
+  enabled: true
+```
+
+The full guide and examples are in the IAM join guide [Joining Services via AWS IAM Role](../../enroll-resources/agents/aws-iam.mdx).
+
+## Use the cluster install script by `cloud-init`
+
+Follow the procedure for [launching an instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-launch-instance-wizard.html)
+and [set the user data](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html?icmpid=docs_ec2_console#userdata-linux).
+The User data field is located in the Advanced details section of the launch instance wizard.
+Enter your cloud-init directive text in the User data field, and then complete the instance launch procedure.
```

Comment:
How essential is it to this guide that the user add the user data via AWS Console? Would it make more sense to structure the guide so that the reader knows they are adding user data to an instance, and has the choice over how they'll deploy the instance?